components/script/dom/globalscope.rs

```diff
@@ -294,15 +294,15 @@ pub struct GlobalScope {
     ///
     /// <https://html.spec.whatwg.org/multipage/#about-to-be-notified-rejected-promises-list>
     #[ignore_malloc_size_of = "mozjs"]
-    uncaught_rejections: DomRefCell<Vec<Box<Heap<*mut JSObject>>>>,
```

Comment:
What was the error that led you to removing the `Box` here?

Reply:
It was saying box is unnecessary 
https://rust-lang.github.io/rust-clippy/master/index.html#/vec_box

Reply:
I think for now you should just allow the warning for `uncaught_rejections` and `consumed_rejections` below. It seems that `Heap` always comes boxed and it's tricky to undo that. Instead put a comment like:

```rust
/// TODO: `Vec` already stores members in a `Box` so this extra box is an unnecessary layer of
/// indirection. Can it be removed?
```

cc @sagudev